gr-digital/lib/constellation_decoder_cb_impl.cc

```diff
@@ -55,6 +65,8 @@ int constellation_decoder_cb_impl::general_work(int noutput_items,
     gr_complex const* in = (const gr_complex*)input_items[0];
     unsigned char* out = (unsigned char*)output_items[0];
 
+    gr::thread::scoped_lock l(d_mutex);
```

Comment:
Place scoped locks in a scope more specific than a function, unless everything in the function needs to be in the critical section, e.g.,
```
general_work()
{
  in = ...
  {
      scoped_lock ...
      functions_that_care_about_locking();
  }
  functions_that_dont_care_about_locking();
}
```
so  the lock is not held for longer than needed.

Reply:
I placed the lock to avoid situations where constellation would change while in the work function.
If it keeps the same dimensionality, there wouln't be an issue. But if it does, then `d_dim` and/or `d_bps` can change.
and since they are used as steps to scan oven the I/O buffers, it could lead to reading outside the buffers memory.

There is basically just one function call in the work function, and it's that one that need protecting.

If we need to avoid locking, we could, but that would restrict the callback to change to a constellation with same dim.

Reply:
Ok, I didn't see the `d_dim` conflict.